aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -279,6 +278,66 @@ class CAFFE2_API OperatorHandle final {
   std::list<Dispatcher::OperatorDef>::iterator operatorIterator_;
 };
 
+/**
+ * This is a handle to an operator schema registered with the dispatcher.
+ * It holds the same information as an OperatorHandle, but it is templated
+ * on the operator arguments and allows calling the operator in an
+ * unboxed way.
+ */
+template<class FuncType>
+class CAFFE2_API TypedOperatorHandle final {
```

Comment:
Suggestion: have this inherit publicly from OperatorHandle, and then just add the typed caller as a new method. This would save you a bunch of boilerplate. I know the modern OO mantra is "prefer composition over inheritance", but TypedOperatorHandle truly fulfills the "is-a" relationship here!